### PR TITLE
Bump vcpkg packages and remove SDL version lock

### PR DESCRIPTION
**Pull request by TheCycoONE**

### CODE REVIEW

**Comment by TheCycoONE** [comment]
Requesting that someone with windows run the versions generated by this PR. The SDL issue should be resolved upstream but it only manifested in Windows.

**Review by lewri — APPROVED** [review]
For windows: Both working fine. Music playback is all fine too (not sure if relevant)

It seems a shame we can't autograb the latest vcpkg as it comes, but I'm guessing that's because we have our overrides.

File: vcpkg-configuration.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     {
       "kind": "git",
       "repository": "https://github.com/CorsixTH/vcpkg-registry",
-      "baseline": "30e94d20cb95de2cc64941c0ba6bd901890ccec6",
+      "baseline": "f8112bf320835623dd837725c85e29a11087135d",
       "packages": [ "ffmpeg", "sdl2-mixer", "luafilesystem", "lpeg" ]
```
[the remark this answers is not in the document]
Comment (TheCycoONE) [reply vcpkg-configuration.json:17]: It will not come for SDL 2. We can probably drop this once we transition to SDL 3 (not yet)

**Comment by TheCycoONE** [comment]
Locking dependencies is important, otherwise we're likely to get broken. Actually appveyor isn't locking and that's why it broke.

**Pull request merged**